### PR TITLE
fix: [CDS-50938]: destination step added for PLG motion

### PR DESCRIPTION
**Pull request by satyam-harness**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/75-cd/pages/get-started-with-cd/CDOnboardingUtils.tsx [hunk 1]
```diff
@@ -327,6 +418,122 @@ export const newDelegateState = {
...
+        tlsClientConfig: { insecure: true },
+        clusterConnectionType: 'USERNAME_PASSWORD'
+      },
+      server,
+      name,
+      namespaces,
+      project
+    }
+
+    clusterPayload = { ...clusterPayload }
```
Comment (ishantmehtaharness) [inline src/modules/75-cd/pages/get-started-with-cd/CDOnboardingUtils.tsx:440]: Why do we need a spread operator here?
We can directly return clusterPayload

File: src/modules/75-cd/pages/get-started-with-cd/CDOnboardingUtils.tsx [hunk 2]
```diff
@@ -327,6 +418,122 @@ export const newDelegateState = {
...
+  clusterData?: ClusterInterface
+  name?: string
+}
+
+export const getAppPayload = (props: PayloadInterface) => {
+  const { repositoryData, clusterData, name } = props
+  return {
+    application: {
+      kind: 'Application',
+      apiVersion: 'argoproj.io/v1alpha1',
```
Comment (ishantmehtaharness) [inline src/modules/75-cd/pages/get-started-with-cd/CDOnboardingUtils.tsx:497]: Is this hardcoded?

File: src/modules/75-cd/pages/get-started-with-cd/CDOnboardingUtils.tsx [hunk 3]
```diff
@@ -327,6 +418,122 @@ export const newDelegateState = {
...
+          ]
+        },
+        source: {
+          repoURLType: repositoryData?.type,
+          revisionType: repositoryData?.revisionType,
+          targetRevision: repositoryData?.targetRevision,
+          repoURL: repositoryData?.repo,
+          path: repositoryData?.path,
+          kustomize: {
+            images: ['quay.io/dexidp/dex:v2.23.0']
```
Comment (ishantmehtaharness) [inline src/modules/75-cd/pages/get-started-with-cd/CDOnboardingUtils.tsx:524]: This is hardcoded in GitOps repo?

**Review by ishantmehtaharness — APPROVED** [review]
Looks Good

**Comment by satyam-harness** [comment]
retrigger jest

**Comment by satyam-harness** [comment]
retrigger jest

File: src/modules/75-cd/pages/get-started-with-cd/CDOnboardingUtils.tsx (hunk 2 above)
Comment (satyam-harness) [reply src/modules/75-cd/pages/get-started-with-cd/CDOnboardingUtils.tsx:497]: yes

File: src/modules/75-cd/pages/get-started-with-cd/CDOnboardingUtils.tsx (hunk 3 above)
Comment (satyam-harness) [reply src/modules/75-cd/pages/get-started-with-cd/CDOnboardingUtils.tsx:524]: yes

**Pull request merged**